pyfar/io/io.py

```diff
@@ -461,6 +463,320 @@ def default_audio_subtype(format):
+        new_shape.append(len(parameters[search_key]))
+    shape.append(len(domain_data))
+    new_shape.append(len(domain_data))
+
+    # Create paired parameter
+    pairs = np.meshgrid(*[x for x in parameters.values()])
+    parameters_pairs = parameters.copy()
+    for idx, key in enumerate(parameters):
+        parameters_pairs[key] = np.transpose(pairs[idx]).flatten()
+
+    # loop over all data and write it into final shape
+    data_expressions = raw_data[:, -n_entries:]
+    data_out = np.empty(shape, dtype=dtype)
+    data_out[:] = np.nan
+    data_inconsistent = False
+    for parameters_idx in range(shape[2]):
+        for i_expression, search_expression in enumerate(expressions):
+            for i_domain, search_domain_value in enumerate(domain_data):
+                expression_idxes = expressions_header == search_expression
+                domain_idxes = domain_header == search_domain_value
+                parameter_idxes = np.array(domain_idxes | True)
+                for key in parameters:
+                    parameter_idxes &= (parameter_header[key] == parameters_pairs[key][parameters_idx])
+                idxes = parameter_idxes & domain_idxes & expression_idxes
+                if any(idxes):
+                    data_out[:, i_expression, parameters_idx, i_domain] \
+                        = data_expressions[:, idxes].flatten()
+                else:
+                    data_inconsistent = True
+                    warnings.warn(r'Parameter data is inconsistent. Missing \
+                        data is filled with nans.')
+
+    # reshape data to final shape
+    if not data_inconsistent:
+        data_out = np.reshape(data_out, new_shape)
+
+    # create object
+    comment = ', '.join(
+        ' '.join(x) for x in zip(all_expressions, units))
+    if domain == 'freq':
+        data = FrequencyData(
+            data_out, domain_data, dtype=dtype,
+            comment=comment)
+    else:
+        data = TimeData(
+            data_out, domain_data, comment=comment)
+
+    # create coordinates
+    if n_dimension > 0:
+        coords_data = raw_data[:, 0:n_dimension].astype(np.float)
+        x = coords_data[:, 0]
+        y = coords_data[:, 1] if coords_data.shape[1] > 1 else np.zeros(
+            coords_data[:, 0].shape)
+        z = coords_data[:, 2] if coords_data.shape[1] > 2 else np.zeros(
+            coords_data[:, 0].shape)
+        coordinates = Coordinates(x, y, z)
+        return data, coordinates
+    return data, None
+
+
+def read_comsol_header(filename):
+    """Returns header information from exported data from COMSOL.
+
+    Parameters
+    ----------
+    filename : str, Path
+        Input file. Excepted input files are '.txt', '.dat' and '.csv'. '.csv'
+        Files are strongly recommended, since .txt and .dat are inconvenient
+        in their format.
+
+    Returns
+    -------
+    expressions : list[str]
+        This list of expressions defines the output expressions. An example
```

Comment:
As above, I would suggest to briefly instroduce the concepts of expressions and parameters.